core/net/ip/resolv.c

```diff
@@ -263,17 +299,65 @@ struct namemap {
 #if RESOLV_CONF_SUPPORTS_MDNS
   int is_mdns:1, is_probe:1;
 #endif
+  /* This stores the hostname for mDNS. */
   char name[RESOLV_CONF_MAX_DOMAIN_NAME_SIZE + 1];
 };
 
-#ifndef UIP_CONF_RESOLV_ENTRIES
-#define RESOLV_ENTRIES 4
-#else /* UIP_CONF_RESOLV_ENTRIES */
-#define RESOLV_ENTRIES UIP_CONF_RESOLV_ENTRIES
-#endif /* UIP_CONF_RESOLV_ENTRIES */
-
+/*
+ * Array of cached hostnames (and service names if using DNS-SD) for other
+ * devices.
+ */
 static struct namemap names[RESOLV_ENTRIES];
 
+#if RESOLV_CONF_SUPPORTS_MDNS
+#if RESOLV_CONF_SUPPORTS_DNS_SD
```

Comment:
Further down below you seem to prefer `#if RESOLV_CONF_SUPPORTS_MDNS && RESOLV_CONF_SUPPORTS_DNS_SD`. I suggest to be consistent by using either or the other.